### PR TITLE
Add CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by ianlewis**
Updates #617 

Signed-off-by: Ian Lewis <ianlewis@google.com>

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+
+These linters will also run as GitHub checks for pull requests via
+[pre-submit.lint.yml](.github/workflows/pre-submit.lint.yml) file.
+
+## Community
+
+This project is part of the [SLSA Community](https://slsa.dev/community) working
+with the SLSA Tooling SIG.
+
+### Communication
```
Comment (laurentsimon) [inline CONTRIBUTING.md:144]: Maybe link to the RELEASE.md somewhere too?

**Review by laurentsimon — APPROVED** [review]
Thanks!

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+## Contributor Guide
+
+This guide will help you understand the overall organization of the
+[slsa-github-generator](https://github.com/slsa-framework/slsa-github-generator)
+project, and direct you to the best places to get started contributing. You'll
+be able to pick up issues, write code to fix them, and get your work reviewed
+and merged.
+
+This document is the single source of truth for how to contribute to the code
+base. Feel free to browse the [open issues] and file new ones, all feedback
```
Comment (asraa) [inline CONTRIBUTING.md:26]:
```suggestion
base. Feel free to browse the [open issues](https://github.com/slsa-framework/slsa-github-generator/issues) and file new ones, all feedback
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+
+#### Create a GitHub account
+
+Before you get started, you will need to [signup](http://github.com/signup) for a GitHub user account.
+
+#### Sign the CLA
+
+> TODO: is this needed?
+
+Before you can contribute to Kubernetes, you will need to sign the
```
Comment (asraa) [inline CONTRIBUTING.md:42]:
```suggestion
Before you can contribute to the SLSA Framework organization, you will need to sign the
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+Before submitting code, you should first complete the following
+prerequisites.
+
+#### Create a GitHub account
+
+Before you get started, you will need to [signup](http://github.com/signup) for a GitHub user account.
+
+#### Sign the CLA
+
+> TODO: is this needed?
```
Comment (asraa) [inline CONTRIBUTING.md:40]: slsa doesn't mention one, I'm not so sure it is:
https://github.com/slsa-framework/slsa/blob/main/CONTRIBUTING.md

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+
+Before you can contribute to Kubernetes, you will need to sign the
+[Contributor License Agreement].
+
+#### Code of Conduct
+
+> TODO: Is there a CoC for the OpenSSF that we can point to?
```
Comment (asraa) [inline CONTRIBUTING.md:47]: https://github.com/ossf/wg-best-practices-os-developers/blob/main/code-of-conduct.md

seems like most repos have this one

File: CONTRIBUTING.md (hunk 2 above)
Comment (ianlewis) [reply CONTRIBUTING.md:26]: This used the link defined at the bottom. I copied this from the Kubernetes contributor guide, in which this was referring to issues related to the contributing guide itself. I think I want to reword this section a bit.

File: CONTRIBUTING.md (hunk 4 above)
Comment (ianlewis) [reply CONTRIBUTING.md:40]: Yeah. I'm not sure exactly what the relationship is for OpenSSF which may end up requiring one. I can maybe leave this section off until we have a more definitive requirement.

File: CONTRIBUTING.md (hunk 5 above)
Comment (ianlewis) [reply CONTRIBUTING.md:47]: I'll add a link here: https://github.com/slsa-framework/slsa/blob/main/code-of-conduct.md

**Pull request merged**